### PR TITLE
fix integer value overflow and add test

### PR DESCRIPTION
**Pull request by nishantmonu51**

### CODE REVIEW

File: server/src/test/java/io/druid/server/coordination/CostBalancerStrategyTest.java [hunk 1]
```diff
@@ -101,12 +101,16 @@ public void setupDummyCluster(int serverCount, int maxSegments)
...
-    // Not using EasyMock as it hampers the performance of multithreads.
-    DataSegment segment = new DataSegment(
-        "DUMMY", day, String.valueOf(index), Maps.<String, Object>newConcurrentMap(),
-        Lists.<String>newArrayList(), Lists.<String>newArrayList(), null, 0, index * 100L
-    );
-    return segment;
+   return getSegment(index,"DUMMY", day);
+  }
+
+  private DataSegment getSegment(int index,String dataSource, Interval interval){
```
Comment (xvrl) [inline server/src/test/java/io/druid/server/coordination/CostBalancerStrategyTest.java]: Formatting

File: server/src/test/java/io/druid/server/coordination/CostBalancerStrategyTest.java [hunk 2]
```diff
@@ -156,4 +160,32 @@ public void testBenchmark() throws InterruptedException
     System.err.println("Latency - Single Threaded (ms): " + latencySingleThread);
     System.err.println("Latency - Multi Threaded (ms): " + latencyMultiThread);
   }
+
+    @Test
+    public void testComputeJointSegmentCost()
+    {
+      DateTime referenceTime = DateTime.now(DateTimeZone.UTC);
```
Comment (xvrl) [inline server/src/test/java/io/druid/server/coordination/CostBalancerStrategyTest.java]: I would prefer if we used specific timestamps in tests to ensure they are deterministic.

**Comment by xvrl** [comment]
Looks good, do you mind squashing your commits and I'll merge

**Comment by nishantmonu51** [comment]
squashed. :)

**Pull request merged**